lib/Reader/readerir.cpp

```diff
@@ -5994,8 +5994,36 @@ void GenIR::switchOpcode(IRNode *Opr) {
   BasicBlock *CurrBlock = LLVMBuilder->GetInsertBlock();
   TerminatorInst *TermInst = CurrBlock->getTerminator();
   SwitchInst *SwitchInstruction = cast<SwitchInst>(TermInst);
-
   SwitchInstruction->setCondition(Opr);
+
+  llvm::Type* OprType = Opr->getType();
+  LLVMContext &LLVMContext = *this->JitContext->LLVMContext;
+  Type *Int32Ty = Type::getInt32Ty(LLVMContext);
+  if (OprType != Int32Ty) {
+    // When we made the case values we assumed int32 as the type for the 
+    // values. But now we have an operand of a different type. LLVM 
+    // requires the values and the switch value to have the same type.
+    // So change the type of the values to be the type of the selector.
+    assert(OprType->isIntegerTy() && "Switch selector is not an integer type");
+    llvm::IntegerType* OprIntType = (llvm::IntegerType*)OprType;
+
```

Comment:
Please use 
IntegerType* OprIntType = cast<IntegerType>(OprType);
instead of the assert and () cast.